### PR TITLE
STAR properly set dbkey on outputs

### PR DESCRIPTION
**Pull request by natefoo**
Like #1902 and #1927 but for STAR.

### CODE REVIEW

File: tools/rgrnastar/macros.xml [hunk 1]
```diff
@@ -17,4 +17,23 @@
          <param argument="--sjdbGTFfile" type="data" format="gff3,gtf" label="Gene model (gff3,gtf) file for splice junctions" optional="true" help="Exon junction information for mapping splices"/>
          <param argument="--sjdbOverhang" type="integer" min="1" value="100" label="Length of the genomic sequence around annotated junctions" help="Used in constructing the splice junctions database. Ideal value is ReadLength-1"/>
     </xml>
+    <xml name="dbKeyActions">
+        <actions>
+            <conditional name="refGenomeSource.geneSource">
+                <when value="indexed">
+                    <action type="metadata" name="dbkey">
+                        <option type="from_data_table" name="rnastar_index2" column="1" offset="0">
```
Comment (martenson) [inline tools/rgrnastar/macros.xml]: was the name of this option supposed to change? (`rnastar_index` vs `rnastar_index2`)

Comment (natefoo) [reply tools/rgrnastar/macros.xml]: Yes, this is what was preventing it from working. The rest is just deduplication.

**Comment by natefoo** [comment]
Well, I've restarted the failing test multiple times and it continues to fail, but I can't figure how I would've caused the error (locking problems with Galaxy SQLite database).

**Comment by jennaj** [comment]
Thanks Nate and hope we can work this out. The tool needs an update on Main anyway (we are behind the toolshed several revs). Be best if has this fix. usegalaxy.eu will want to update all three tools, too, they are also behind the MTS for this one and maybe other two (didn't check those) but am not sure of their update process. For some reason, I thought they had some extra magic that kept things current with MTS :)

**Comment by mblue9** [comment]
I tested this locally and 2/3 of the failing tests here pass fine for me locally with 18.05. But one (the 5th test) this one:
```
mkdir -p tempstargenomedir && STAR --runMode genomeGenerate --genomeDir 'tempstargenomedir' --genomeFastaFiles '/private/var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T/tmp53kbkU/files/000/dataset_25.dat' --runThreadN ${GALAXY_SLOTS:-4} &&  STAR --runThreadN ${GALAXY_SLOTS:-4} --genomeLoad NoSharedMemory --genomeDir 'tempstargenomedir'  --readFilesIn '/private/var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T/tmp53kbkU/files/000/dataset_24.dat'   --readFilesCommand zcat   --outSAMtype BAM SortedByCoordinate    --outSAMattributes All --outFilterIntronMotifs None --outFilterIntronStrands RemoveInconsistentStrands  --outReadsUnmapped None --chimSegmentMin 12 --chimJunctionOverhangMin 12 --alignSJDBoverhangMin 10 --alignMatesGapMax 200000 --alignIntronMax 200000 --twopassMode Basic --twopass1readsN -1 --chimSegmentReadGapMax 3 --alignSJstitchMismatchNmax 5 -1 5 5 --outSAMstrandField intronMotif
```
also fails for me but with the error:
```
SamtoolsError: 'samtools returned with error 1: stdout=, stderr=samtools index: "/private/var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T/tmp53kbkU/files/000/dataset_29.dat" is in a format that cannot be usefully indexed\n'
```

**Comment by mvdbeek** [comment]
Meh, we should have a jenkins job that test all of our tools before a new galaxy release. I've had something like that cooked up at the GCC in Norwich, maybe it's time to undust that.

**Comment by natefoo** [comment]
Thanks @mblue9 and @mvdbeek!

**Comment by bgruening** [comment]
Thanks all!

**Comment by mblue9** [comment]
>we should have a jenkins job that test all of our tools before a new galaxy release. I've had something like that cooked up at the GCC in Norwich, maybe it's time to undust that.

@mvdbeek I think this is such a great idea!!! 

And I would definitely appreciate a heads up on finding out if any tools I've been working on are broken with a new release so I can try to proactively fix (rather than find out when I'm submitting PRs) 👍 👍

**Pull request merged**